### PR TITLE
Add delete and peek functionality to dashboard

### PR DESCRIPTION
**Pull request by rshermanTHG**
### Motivation

Add some administration functionality to the dashboard. 

1. Ability to peek at the contents of messages in a subscriptions backlog
2. Able to clear the backlog in a subscription
3. Able to delete subscriptions that have no consumers
4. Able to delete a namespace that has no topics

### Modifications

All modifications are to the dashboard module only.

- Added a new page that lists messages in the backlog as a link. When this link is clicked on the message pops up in a modal dialog
- Added a deleted flag to several models and added this flag to the appropriate filters. (when an object is deleted this will keep the dashboard reflective of the state of the brokers until the next collector run).
- Added functionality to delete a subscription if there are no consumers. Marks the subscription as deleted in the database. If this leaves a topic with no subscriptions then the topic is also marked in the database as deleted.
- Added functionality to delete a namespace if there are no topics for that namespace. Marks the namespace as deleted in the database.

### CODE REVIEW

**Comment by jiazhai** [comment]
@tuteng would you please also help review it?

**Comment by jiazhai** [comment]
run java8 tests

File: dashboard/django/collector.py [hunk 1]
```diff
@@ -324,6 +325,19 @@ def _fetch_broker_stats(cluster, broker_host_port, timestamp):
             replication.topic = replication.topic
             replication.save()
 
+    tenants = get(broker_url, '/admin/v2/tenants')
+    for tenant_name in tenants:
+        namespaces = get(broker_url, '/admin/v2/namespaces/' + tenant_name)
+        for namespace_name in namespaces:
+            logger.error('tn - ' + tenant_name + ' ns - ' + namespace_name)
```
Comment (tuteng) [inline dashboard/django/collector.py]: logger.info can be used here.

File: dashboard/django/stats/views.py [hunk 2]
```diff
@@ -120,6 +132,15 @@ def namespace(request, namespace_name):
     })
 
 
+def deleteNamespace(request, namespace_name):
+    url = settings.SERVICE_URL + '/admin/v2/namespaces/' + namespace_name
```
Comment (tuteng) [inline dashboard/django/stats/views.py]: This option settings.SERVICE_URL needs to be included in the configuration file settings.

File: dashboard/django/stats/views.py [hunk 3]
```diff
@@ -282,3 +305,55 @@ def clusters(request):
     return render(request, 'stats/clusters.html', {
         'clusters' : clusters,
     })
+
+
+def clearSubscription(request, topic_name, subscription_name):
+    url = settings.SERVICE_URL + '/admin/v2/' + topic_name + '/subscription/' + subscription_name + '/skip_all'
```
Comment (tuteng) [inline dashboard/django/stats/views.py]: This option settings.SERVICE_URL needs to be included in the configuration file settings.

File: dashboard/django/stats/views.py [hunk 4]
```diff
@@ -282,3 +305,55 @@ def clusters(request):
...
+
+    return render(request, 'stats/messages.html', {
+        'topic' : topic,
+        'subscription' : subscription,
+        'title' : topic.name,
+        'subtitle' : subscription_name,
+    })
+
+def peek(request, topic_name, subscription_name, message_number):
+    url = settings.SERVICE_URL + '/admin/v2/' + topic_name + '/subscription/' + subscription_name + '/position/' + message_number
```
Comment (tuteng) [inline dashboard/django/stats/views.py]: This option settings.SERVICE_URL needs to be included in the configuration file settings.

File: dashboard/django/stats/views.py [hunk 5]
```diff
@@ -282,3 +305,55 @@ def clusters(request):
...
     })
+
+
+def clearSubscription(request, topic_name, subscription_name):
+    url = settings.SERVICE_URL + '/admin/v2/' + topic_name + '/subscription/' + subscription_name + '/skip_all'
+    requests.post(url)
+    return redirect('topic', topic_name=topic_name)
+
+def deleteSubscription(request, topic_name, subscription_name):
+    url = settings.SERVICE_URL + '/admin/v2/' + topic_name + '/subscription/' + subscription_name
```
Comment (tuteng) [inline dashboard/django/stats/views.py]: This option settings.SERVICE_URL needs to be included in the configuration file settings.

File: dashboard/django/collector.py (hunk 1 above)
Comment (rshermanTHG) [reply dashboard/django/collector.py]: Removed as unnecessary.

File: dashboard/django/stats/views.py (hunk 2 above)
Comment (rshermanTHG) [reply dashboard/django/stats/views.py]: Done

File: dashboard/django/stats/views.py (hunk 4 above)
Comment (rshermanTHG) [reply dashboard/django/stats/views.py]: Done

File: dashboard/django/stats/views.py (hunk 5 above)
Comment (rshermanTHG) [reply dashboard/django/stats/views.py]: Done

File: dashboard/django/stats/views.py (hunk 3 above)
Comment (rshermanTHG) [reply dashboard/django/stats/views.py]: Done

**Comment by tuteng** [comment]
run java8 tests

**Comment by jiazhai** [comment]
@rshermanTHG  some files need ASF2 license in header:
Here is a error reported:
```
[WARNING] Missing header in: /home/jenkins/jenkins-slave/workspace/pulsar_precommit_java8/dashboard/django/stats/static/stats/additional.css
```

**Comment by rshermanTHG** [comment]
> @rshermanTHG some files need ASF2 license in header:
> Here is an error reported:
> 
> ```
> [WARNING] Missing header in: /home/jenkins/jenkins-slave/workspace/pulsar_precommit_java8/dashboard/django/stats/static/stats/additional.css
> ```

Sorted

**Comment by sijie** [comment]
@tuteng can you please review this again?

**Comment by tuteng** [comment]
LGTM

**Pull request merged**